extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayProducer.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.flyway.runtime;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import org.flywaydb.core.Flyway;
+import org.flywaydb.core.api.configuration.FluentConfiguration;
+
+import io.agroal.api.AgroalDataSource;
+
+@ApplicationScoped
+public class FlywayProducer {
+    @Inject
+    Instance<AgroalDataSource> dataSource;
```

Comment:
I think, maybe, it could work to remove the `Instance` here now. I recently made the dataSource application scoped instead of singleton and it might fix the issue you had.

Reply:
Obviously, you will need to rebase on top of current master to see if the issue is fixed.

Reply:
Yes, it is working now 👍